src/utils.ts

```diff
@@ -39,29 +39,25 @@ export const getNotesFromPath = async (vault: Vault, path: string): Promise<TFil
   });
 }
 
-export async function getTagsFromNote(file: TFile, vault: Vault): Promise<string[]> {
-  const fileContent = await vault.cachedRead(file);
-  const frontMatter = parseYaml(fileContent.split('---')?.[1] ?? '') || {};
-  const tags = frontMatter.tags || [];
+export async function getTagsFromNote(file: TFile, app: App): Promise<string[]> {
+  const frontMatter = app.metadataCache.getFileCache(file)?.frontmatter
+  const tags = frontMatter?.tags || [];
```

Comment:
For the sake of people who will be using this library, even though Logan seems to not consider the tags in the note content for whatever reason, Obsidian in fact does.

Should be a oneliner-change after this PR, you can take inspiration from here